src/elasticarray.jl

```diff
@@ -57,76 +57,132 @@ ElasticArray(A::AbstractArray{T,N}) where {T,N} = copyto!(ElasticArray{T,N}(unde
 Base.convert(::Type{T}, A::AbstractArray) where {T<:ElasticArray} = A isa T ? A : T(A)
 
 
+Base.similar(::Type{ElasticArray{T}}, dims::Dims{N}) where {T,N} = ElasticArray{T}(undef, dims...)
+
+
+@inline function Base.:(==)(A::ElasticArray, B::ElasticArray)
+    return ndims(A) == ndims(B) && A.kernel_size == B.kernel_size && A.data == B.data
+end
 
 
-import Base.==
-(==)(A::ElasticArray, B::ElasticArray) =
-    ndims(A) == ndims(B) && A.kernel_size == B.kernel_size && A.data == B.data
+@inline Base.size(A::ElasticArray) = (A.kernel_size..., div(length(eachindex(A.data)), A.kernel_length))
 
+@inline Base.length(A::ElasticArray) = length(A.data)
 
-Base.parent(A::ElasticArray) = A.data
+@propagate_inbounds Base.getindex(A::ElasticArray, i::Int) = getindex(A.data, i)
 
-Base.size(A::ElasticArray) = (A.kernel_size..., div(length(eachindex(A.data)), A.kernel_length))
-@propagate_inbounds Base.getindex(A::ElasticArray, i::Integer) = getindex(A.data, i)
-@propagate_inbounds Base.setindex!(A::ElasticArray, x, i::Integer) = setindex!(A.data, x, i)
-@inline Base.IndexStyle(A::ElasticArray) = IndexStyle(A.data)
+@propagate_inbounds Base.setindex!(A::ElasticArray, x, i::Int) = setindex!(A.data, x, i)
 
-Base.length(A::ElasticArray) = length(A.data)
+Base.IndexStyle(::Type{<:ElasticArray}) = IndexLinear()
 
-Base.dataids(A::ElasticArray) = Base.dataids(A.data)
 
 
-@inline function Base.resize!(A::ElasticArray{T,N}, dims::Vararg{Integer,N}) where {T,N}
-    kernel_size, size_lastdim = _split_resize_dims(A, dims)
+@inline Base.resize!(A::ElasticArray{T,N}, dims::Vararg{Integer,N}) where {T,N} = resize!(A, dims)
+
+@inline function Base.resize!(A::ElasticArray{T,N}, dims::NTuple{N,Integer}) where {T,N}
+    _, size_lastdim = _split_resize_dims(A, dims)
     resize!(A.data, A.kernel_length.divisor * size_lastdim)
-    A
+    return A
+end
+
+@inline function Base.resize!(A::ElasticArray, size_lastdim::Integer)
+    resize!(A.data, A.kernel_length.divisor * size_lastdim)
+    return A
+end
+
+
+@inline Base.sizehint!(A::ElasticArray{T,N}, dims::Vararg{Integer,N}) where {T,N} = sizehint!(A, dims)
+
+@inline function Base.sizehint!(A::ElasticArray{T,N}, dims::NTuple{N,Integer}) where {T,N}
```

Comment:
I'd like to stay with `dims::Vararg{Integer,N}` instead of `dims::NTuple{N,Integer}`, it's more in line with Julia array ctor syntax.

Reply:
see below